controls/oscal.py

```diff
@@ -6,56 +6,47 @@
 from pathlib import Path
 import sys
 
+import auto_prefetch
+from django.db import models
+from django.utils.functional import cached_property
+from jsonfield import JSONField
+
+
 CATALOG_PATH = os.path.join(os.path.dirname(__file__), 'data', 'catalogs')
-EXTERNAL_CATALOG_PATH = os.path.join(f"{os.getcwd()}",'local', 'controls', 'data', 'catalogs')
+BASELINE_PATH = os.path.join(os.path.dirname(__file__),'data','baselines')
+
+class CatalogData(auto_prefetch.Model):
+    catalog_key = models.CharField(max_length=100, help_text="Unique key for catalog", unique=True, blank=False, null=False)
+    catalog_json = JSONField(blank=True, null=True, help_text="JSON object representing the OSCAL-formatted control catalog.")
+    baselines_json = JSONField(blank=True, null=True, help_text="JSON object representing the baselines for the catalog.")
+    created = models.DateTimeField(auto_now_add=True, db_index=True)
+    updated = models.DateTimeField(auto_now=True, db_index=True)
+
+    def __str__(self):
+        return "'%s id=%d'" % (self.catalog_key, self.id)
+
+    def __repr__(self):
+        # For debugging.
+        return "'%s id=%d'" % (self.catalog_key, self.id)
 
 class Catalogs(object):
     """Represent list of catalogs"""
 
     # well known catalog identifiers
-
     NIST_SP_800_53_rev4 = 'NIST_SP-800-53_rev4'
     NIST_SP_800_53_rev5 = 'NIST_SP-800-53_rev5'
     NIST_SP_800_171_rev1 = 'NIST_SP-800-171_rev1'
     CMMC_ver1 = 'CMMC_ver1'
 
     def __init__(self):
-        self.catalog_path = CATALOG_PATH
-        # self.catalog = None
         self.catalog_keys = self._list_catalog_keys()
         self.index = self._build_index()
 
-    def extend_external_catalogs(self, catalog_info, extendtype):
-        """
-        Add external catalogs to list of catalogs
-        """
-        os.makedirs(EXTERNAL_CATALOG_PATH, exist_ok=True)
-        external_catalogs = [file for file in os.listdir(EXTERNAL_CATALOG_PATH) if
-                  file.endswith('.json')]
-        catalog_info = check_and_extend(catalog_info, external_catalogs, extendtype, "_catalog")
-
-        return catalog_info
-
-    def _list_catalog_files(self):
-        return self.extend_external_catalogs([
-            'NIST_SP-800-53_rev4_catalog.json',
-            'NIST_SP-800-53_rev5_catalog.json',
-            'NIST_SP-800-171_rev1_catalog.json',
-            'CMMC_ver1_catalog.json'
-        ], "files")
-
     def _list_catalog_keys(self):
-
-        return self.extend_external_catalogs([
-            Catalogs.NIST_SP_800_53_rev4,
-            Catalogs.NIST_SP_800_53_rev5,
-            Catalogs.NIST_SP_800_171_rev1,
-            Catalogs.CMMC_ver1
-        ], "keys")
+        return [item['catalog_key'] for item in CatalogData.objects.order_by().values('catalog_key').distinct()]
```

Comment:
from [item['catalog_key'] for item in CatalogData.objects.order_by().values('catalog_key').distinct()

to

CatalogData.objects.order_by('catalog_key').values_list('catalog_key', flat=True).distinct()

Reply:
Change made.